src/equisolve/vanilla/preprocessing/_base.py

```diff
@@ -0,0 +1,158 @@
+        ```
+
+    :param cell_gradient: string or function that returns array of shape (3, 3)
+        NOT IMPLEMENTED
+
+    """
+
+    if not isinstance(frames, list):
+        frames = [frames]
+    if isinstance(frames[0], ase.Atoms):
+        return _ase_frames_property_to_tensormap(frames, property, positions_gradient, cell_gradient, is_structure_property)
+    elif isinstance(frames[0], chemfiles.Frame):
+        raise NotImplementedError("Extracting porperties out of frame objects of type chemfiles.Frame has not been implemented yet.")
+    else:
+        raise NotImplementedError(f"Extracting porperties out of frame objects of type {type(frames[0])} is not supported.")
+
+
+def _ase_frames_property_to_tensormap(frames, property=None, positions_gradient=None, cell_gradient=None, is_structure_property=True):
+    if cell_gradient is not None:
+        raise NotImplementedError("cell_gradient is not supported at the moment.")
+    if not(is_structure_property):
+        raise NotImplementedError("Support for environment properties has not been implemented yet.")
+
+    if isinstance(property, str):
+        get_property = lambda frame : frame.info[property]
+    else:
+        get_property = property
+    if isinstance(positions_gradient, str):
+        get_positions_gradient = _get_default_positions_gradient_extraction(frames[0], positions_gradient)
+    else:
+        get_positions_gradient = positions_gradient
+    #if isinstance(cell_gradient, str):
+    #    get_stress = lambda frame : -frame.arrays[property]
+
+
+    keys = Labels(
+        names=["idx"],
+        values=np.array([[0]], dtype=np.int32)
+    )
+    #keys = Labels.single()
+
+    if property is None:
+        raise NotImplementedError("Needs to be implemented before PR gets merged.")
+    else:
+        property_values = np.array([get_property(frame) for frame in frames])
+        # -1 dimension because of property dimension
+
+        samples_property = Labels(
+            names=["structure"],
+            values=np.array([
+                [s]
+                for s in range(len(frames))
+            ], dtype=np.int32)
+        )
+
+        # gradients needs sample dimension
+        offsets = np.array([0] + [len(frame) for frame in frames])
+        samples_gradients = Labels(
+            names=["sample", "structure", "atom"],
+            values=np.array([
+                [offsets[s]+a, s, a]
+                for s in range(len(frames))
+                for a in range(len(frames[s]))
+            ], dtype=np.int32)
+        )
+
+        blocks = [TensorBlock(
+            values=property_values.reshape(-1,1),
+            samples=samples_property,
+            components=[],
+            properties=Labels(['property'], np.array([(0,)], dtype=np.int32))
+        )]
+
+    if positions_gradient is not None:
+        positions_gradient  = np.concatenate([get_positions_gradient(frame) for frame in frames], axis=0).reshape(-1, 3, 1)
```

Comment:
so if the positions gradients (or the values) shape is wrong/of the wrong dtype/... this will error when adding the gradient below. Should we add some validation around here instead to be able to give nicer error messages to the users?